lib/src/modules/task.cc

```diff
@@ -592,6 +713,20 @@ ActionResponse Task::callAction(const ActionRequest& request)
 
     task_params.set<std::string>("_task", task_name);
 
+    std::vector<std::string> meta_files;
+    // Only get files array from top level of metatdata instead of puppetserver supplied file data
```

Comment:
This still looks wrong.

Reply:
I guess it works.

Reply:
I experimented a bit with this idea. In most of the unit test examples the files key will not be present. This avoids the leatherman data_type_error exception but leaves an empty object that will have to be guarded against before calling `getMultiFiles` (because it is not of vector<string>). Also I was not finding an easy way of comparing `leatherman::JsonContainer`. How would you compare those? would they end up being the same memory address?

Reply:
Ah, don't necessarily have an answer. Let's look at it later.